pywbem/cim_obj.py

```diff
@@ -1382,6 +1435,138 @@ def tocimxmlstr(self, indent=None):
+            # numbers in addition to single quoted strings, see issue #932.
+            # Pywbem implements only single quoted strings.
+            # TODO 01/18 AM Remove note once charValue is fixed in DSP0207
+
+            cimval = val[1:-1]
+            cimval = re.sub(r'\\(.)', r'\1', cimval)
+            cimval = _ensure_unicode(cimval)
+            if len(cimval) != 1:
+                raise ValueError("WBEM URI has a char16 keybinding with an "
+                                 "incorrect length: %s=%r" % (key, val))
+            return cimval
+
+        if val.lower() in ('true', 'false'):
+            # The key value must be CIM type:
+            # * boolean (see booleanValue in DSP00004)
+            cimval = val.lower() == 'true'
+            return cimval
+
+        if DECIMAL_VALUE.match(val):
+            # The key value must be one of the integer CIM types:
+            # * uintNN, sintNN (see decimalValue in DSP00004)
+
+            # For integer keybindings in an untyped WBEM URI, it is not
+            # possible to detect the exact CIM data type. Therefore, pywbem
+            # stores the value as a Python int type (or long in Python 2,
+            # if needed).
+
+            # Note that DSP0207 only allows only US-ASCII 0-9 for decimal
+            # numbers. The int() function supports all decimal Unicode
+            # digits (e.g. US-ASCII 0-9, ARABIC-INDIC digits, superscripts,
+            # or subscripts) and raises ValueError for non-decimal digits
+            # (e.g. Kharoshthi digits).
+
+            # Therefore, the decimalValue format has been checked explicitly,
+            # and an error in the int() function is not expected.
+
+            cimval = int(val)  # returns long if needed
+            return cimval
+
+        if REAL_VALUE.match(val):
+            # The key value must be one of the real CIM types:
+            # * realNN (see realValue in DSP00004)
+
+            # For real/float keybindings in an untyped WBEM URI, it is not
+            # possible to detect the exact CIM data type. Therefore, pywbem
+            # stores the value as a Python float type.
+
+            # The float() function supports a superset of input formats
+            # compared to the realValue definition in DSP0004 (for example,
+            # "1." is allowed for float() but not for realValue), plus it
+            # has the same support for decimal Unicode digits as int().
+            # Therefore, the US-ASCII 0-9 digits have been checked explicitly,
+            # and an error in the int() function is not expected.
+
+            # Therefore, the realValue format has been checked explicitly,
+            # and an error in the float() function is not expected.
+
+            # Note that the special values 'INF', '-INF', and 'NAN' are
+            # also covered by Python float().
+
+            cimval = float(val)
+            return cimval
+
+        # The only remaining CIM type now is:
```

Comment:
This phrase is confusing.  Took a while to realise that what I think you meant was that all CIM Types are covered???

Reply:
DONE: Changed to "At this point, all CIM types have been processed, except:"